src/main/java/io/hhplus/tdd/point/service/PointService.java

```diff
@@ -0,0 +1,75 @@
+package io.hhplus.tdd.point.service;
+
+import io.hhplus.tdd.database.PointHistoryTable;
+import io.hhplus.tdd.database.UserPointTable;
+import io.hhplus.tdd.point.entity.PointHistory;
+import io.hhplus.tdd.point.entity.UserPoint;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.locks.Lock;
+
+import io.hhplus.tdd.lock.LockManager;
+
+import static io.hhplus.tdd.point.entity.TransactionType.*;
+
+@Service
+@RequiredArgsConstructor
+public class PointService {
+
+    private final UserPointTable userPointTable;
+    private final PointHistoryTable pointHistoryTable;
+    private final LockManager lockManager = new LockManager();
+
+    public UserPoint getUserPointById(long id) {
+        return userPointTable.selectById(id);
+    }
+
+    public List<PointHistory> getPointHistoryByUserId(long userId) {
+        return pointHistoryTable.selectAllByUserId(userId);
+    }
+
+    public UserPoint chargeUserPoint(long userId, long amount) {
+        Lock lock = lockManager.getLockForUser(userId);
+        lock.lock();
+        UserPoint chargedUserPoint;
+        try {
+            UserPoint userPoint = userPointTable.selectById(userId);
+            chargedUserPoint = userPoint.charge(amount);
+            userPointTable.insertOrUpdate(userId, chargedUserPoint.point());
+        } finally {
+            lock.unlock();
+            lockManager.cleanupLock(userId, lock);
+        }
+
+        // 비동기적으로 포인트 히스토리 기록
+        CompletableFuture.runAsync(() -> 
```

Comment:
```suggestion
@Async
public void insert() {}
```
반복되는 동작이라면 함수로 만들어 관리하는건 어떨까요?

추가로 Executor 와 thread pool 에 관해서 학습하면 더 좋을것 같습니다